source/agora/consensus/Validation.d

```diff
@@ -57,18 +57,72 @@ public string isInvalidReason (const Transaction tx,
     Amount sum_unspent;
 
     const tx_hash = hashFull(tx);
-    foreach (input; tx.inputs)
+
+    if (tx.type == TxType.Freeze)
     {
-        // all referenced outputs must be present
-        UTXOSetValue value;
-        if (!findUTXO(input.previous, input.index, value))
-            return "Transaction: Input ref not in UTXO";
+        if (tx.inputs.length != 1)
+            return "Transaction: too many inputs";
+
+        if (tx.outputs.length != 1)
+            return "Transaction: too many outputs";
+
+        foreach (input; tx.inputs)
+        {
+            // all referenced outputs must be present
+            UTXOSetValue utxo_value;
+            if (!findUTXO(input.previous, input.index, utxo_value))
+                return "Transaction: Input ref not in UTXO";
+
+            if (!utxo_value.output.address.verify(input.signature, tx_hash[]))
+                return "Transaction: Input has invalid signature";
+
+            if (!sum_unspent.add(utxo_value.output.value))
+                return "Transaction: Input overflow";
 
-        if (!value.output.address.verify(input.signature, tx_hash[]))
-            return "Transaction: Input has invalid signature";
+            if (utxo_value.type != TxType.Payment)
+                return "Transaction: Only available freeze on TxType.Payment";
 
-        if (!sum_unspent.add(value.output.value))
-            return "Transaction: Input overflow";
+            if (utxo_value.output.value != Amount(40_000))
```

Comment:
The unit is 10^7 so you're missing a few 0

Reply:
Oh, I see. I have to fix it.

Reply:
I think we need to increase the amount of Genesis Block Coins.